src/binary-reader-linker.cc

```diff
@@ -314,12 +263,8 @@ Result BinaryReaderLinker::OnExport(uint32_t index,
 }
 
 Result BinaryReaderLinker::OnFunctionName(uint32_t index, StringSlice name) {
-  while (binary->debug_names.size() < index) {
-    binary->debug_names.emplace_back();
-  }
-  if (binary->debug_names.size() == index) {
-    binary->debug_names.push_back(string_slice_to_string(name));
-  }
+  assert(index < binary->debug_names.size());
```

Comment:
I was mistaken in my initial comment. Error rather than assert sounds good to me though!